src/hooks/useMaciSignup.ts

```diff
@@ -33,6 +34,8 @@ export const useMaciSignup = (): IUseMaciSignUpData => {
     return values?.[length - 1]?.id;
   }, [attestations]);
 
+  const isAllowedToVote = useMemo(() => Boolean(attestationId), [attestationId]);
```

Comment:
Thinking about do we need this? 
In my opinion, we could use `isRegistered` to check if the user is allowed to vote or not, and if the user does not have an attestation, we actually should disable `signup` instead of `vote`?

Reply:
It's not a good UX pattern to disable final states, it's better to show what is the problem because disabled buttons don't tell anything for user.

Reply:
>  if the user does not have an attestation, we actually should disable `signup` instead of `vote`?

I think I agree with you but wouldn't we still want this kind of eligibility boolean to determine if a user is eligible to vote or not? 

If a user isn't registered, they should not be able to vote.

If a user isn't eligible, they should not be able to register or vote.

Reply:
> If a user isn't registered, they should not be able to vote.
> If a user isn't eligible, they should not be able to register or vote.
--> agree with that.

I added an error modal to tell the user why they cannot register in PR: https://github.com/privacy-scaling-explorations/maci-rpgf/pull/23